utils/roadie-backstage-entity-validator/src/validator.test.js

```diff
@@ -666,4 +666,94 @@ spec:
       );
     });
   });
+
+  describe('source-location annotation', () => {
+    const baseEntity = `
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: test-entity
+  description: Test entity
+  annotations:
+    backstage.io/source-location: %s
+spec:
+  type: service
+  owner: user:test
+  lifecycle: experimental
+`;
+
+    it('should validate URLs with trailing slash for directories', async () => {
+      const validUrls = [
+        'url:https://github.com/org/repo/tree/main/',
+        'gitlab:https://gitlab.com/org/repo/-/tree/main/',
+        'github:https://github.com/org/repo/tree/main/',
+        'azure/api:https://dev.azure.com/org/project/_git/repo?path=/',
+        'dir:path/to/dir/',
+      ];
+
+      for (const url of validUrls) {
```

Comment:
FYI Jest has native support for this.

```suggestion
    it.each([
      'url:https://github.com/org/repo/tree/main/',
      'gitlab:https://gitlab.com/org/repo/-/tree/main/',
      'github:https://github.com/org/repo/tree/main/',
      'azure/api:https://dev.azure.com/org/project/_git/repo?path=/',
      'dir:path/to/dir/',
    ])(should validate URLs with trailing slash for directories: %s', async (url) => {
```